lib/routes/account.js

```diff
@@ -1021,50 +1045,78 @@ module.exports = function (
         validate: {
           payload: {
             uid: isA.string().max(32).regex(HEX_STRING).required(),
-            code: isA.string().min(32).max(32).regex(HEX_STRING).required()
+            code: isA.string().min(32).max(64).regex(HEX_STRING).required()
           }
         }
       },
       handler: function (request, reply) {
-        log.begin('Account.RecoveryEmailVerify', request)
         var uid = request.payload.uid
         var code = Buffer(request.payload.code, 'hex')
-        db.account(Buffer(uid, 'hex'))
-          .then(
-            function (account) {
-              // If the account is already verified, they may be e.g.
-              // clicking a stale link.  Silently succeed.
-              if (account.emailVerified) {
-                log.increment('account.already_verified')
-                return true
-              }
-              if (!butil.buffersAreEqual(code, account.emailCode)) {
-                throw error.invalidVerificationCode()
-              }
-              log.timing('account.verified', Date.now() - account.createdAt)
-              log.event('verified', { email: account.email, uid: account.uid, locale: account.locale })
-              log.increment('account.verified')
 
-              // send a push notification to all devices that the account changed
-              push.notifyUpdate(uid)
+        if(code.length === 32) {
```

Comment:
When I call this endpoint with the code that was sent in the email, I see the following error in my auth-server console:


> request.summary {"op":"request.summary","code":500,"errno":999,"rid":"1461185740584:Shanes-Retina-MBP.local:18620:in9btvlk:10026","path":"/v1/recovery_email/verify_code","lang":"en-US,en;q=0.5","agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:46.0) Gecko/20100101 Firefox/46.0","remoteAddressChain":["127.0.0.1"],"t":1,"uid":"d6c8a4310117427080db8bdfc7884a3d","stack":"TypeError: Uncaught error: Cannot convert undefined or null to object\n    at Function.keys (native)\n    at unbuffer (/Users/stomlinson/development/fxa-auth-server/lib/crypto/butil.js:36:21)\n    at DB.verifyToken (/Users/stomlinson/development/fxa-auth-server/lib/db.js:764:9)\n    at routes.handler (/Users/stomlinson/development/fxa-auth-server/lib/routes/account.js:1061:14)\n    at Object.internals.handler (/Users/stomlinson/development/fxa-auth-server/node_modules/hapi/lib/handler.js:93:36)\n    at /Users/stomlinson/development/fxa-auth-server/node_modules/hapi/lib/handler.js:28:23\n    at internals.Protect.run (/Users/stomlinson/development/fxa-auth-server/node_modules/hapi/lib/protect.js:56:5)\n    at exports.execute (/Users/stomlinson/development/fxa-auth-server/node_modules/hapi/lib/handler.js:22:22)\n    at /Users/stomlinson/development/fxa-auth-server/node_modules/hapi/lib/request.js:370:13\n    at iterate (/Users/stomlinson/development/fxa-auth-server/node_modules/hapi/node_modules/items/lib/index.js:35:13)\n    at done (/Users/stomlinson/development/fxa-auth-server/node_modules/hapi/node_modules/items/lib/index.js:27:25)\n    at /Users/stomlinson/development/fxa-auth-server/node_modules/hapi/node_modules/hoek/lib/index.js:841:22\n    at nextTickCallbackWith0Args (node.js:419:9)\n    at process._tickDomainCallback (node.js:389:13)"}

The function signature of [verifyToken expects 2 parameters, a tokenVerificationId and a token](https://github.com/mozilla/fxa-auth-server/pull/1232/files#diff-be228df81b99a384aa50ad50bde7fca0R761)